### ISSUE TITLE
Optional parameters (opt) are overridden when initializing Emitter with {host, port} object.

### ISSUE THREAD

**Issue by xprudhomme** [socketio/socket.io-emitter#95]
**Issue**
When trying to initialize the Emitter with an `{host, port}` object and the `key` option, I noticed nothing was published as expected on Redis, by running the Redis monitor: the `key` used remains the same as the default one `socket.io` : 

`
1615484908.414247 [0 127.0.0.1:58230] "publish" "socket.io#/waiting_lines#" "\x93\xa7emitter\x83\xa4type\x02\xa4data\x92\xa4list\x84\xa4data\x94\x8d\xa2id\xd9$1eb84855-3081-4067-90a5-6bb2cb5e20d6\xaacreated_at\xb82020-09-23T11:44:08.000Z\xaaupdated_at\xb82021-01-29T06:33:42.000Z
`

**I have the following packages:**
`    "socket.io": "^2.3.0",
    "socket.io-emitter": "^3.2.0",
    "socket.io-redis": "^5.4.0",
`

**I initialized the Emitter, as follows:**
```javascript
import ioEmitter, { SocketIOEmitter } from "socket.io-emitter";
const io = ioEmitter({host: '127.0.0.1', port: 6379 }, { key: "some-socket.io-key" });
```

What I've noticed in `socket.io-emitter/index.js` file, in the `init` function (adding some console.log for debug purposes):
```javascript
/**
 * Socket.IO redis based emitter.
 *
 * @param {Object} redis client (optional)
 * @param {Object} options
 * @api public
 */

function init(redis, opts){
  console.log(` [SOCKET.IO-EMITTER] Options: `, opts);
  opts = opts || {};

  console.log(` [SOCKET.IO-EMITTER] Options (2): `, opts);

  if ('string' == typeof redis) {
    redis = client(redis);
  }

  if (redis && !redis.hset) {
    opts = redis;
    redis = null;
  }

  console.log(` [SOCKET.IO-EMITTER] Options (3): `, opts);

  if (!redis) {
    if (!opts.socket && !opts.host) throw new Error('Missing redis `host`');
    if (!opts.socket && !opts.port) throw new Error('Missing redis `port`');
    redis = opts.socket
      ? client(opts.socket)
      : client(opts.port, opts.host);
  }

  var prefix = opts.key || 'socket.io';
  console.log(` [SOCKET.IO-EMITTER] Prefix used: `, prefix);
  return new Emitter(redis, prefix, '/');
}
```

**Here's what the console shown:**
` [SOCKET.IO-EMITTER] Options:  { key: 'some-socket.io-key' }`
` [SOCKET.IO-EMITTER] Options (2):  { key: 'some-socket.io-key' }`
` [SOCKET.IO-EMITTER] Options (3):  { host: '127.0.0.1', port: 6379 }`
` [SOCKET.IO-EMITTER] Prefix used:  socket.io`

**To conclude with:**
Basically , I cannot config a custom key, as the opts object is overriden. I suppose this is not how it's supposed to work, right?

**Workaround**
To be able to get it working as expected, I had to do as follows:
```typescript
    let redisOptions: SocketIORedisOptions = {
      host,
      port,
    };
    // Unable to set key prop in previous object as TypeScript 
    // would throw an error saying 'key' does not exist in SocketIORedisOptions type 
    redisOptions["key"] = "some-socket.io-key";
    const io = ioEmitter(redisOptions);
```

**Comment by darrachequesne** [comment]
Yes, this is for this kind of problems that the redis client won't be created by the library anymore (https://github.com/socketio/socket.io-emitter/commit/a70db12877d901dd0f7085def0a91145b7c83163)

Before (`socket.io-emitter@3`):

```js
const io = require("socket.io-emitter")({ host: "127.0.0.1", port: 6379 });
```

After (`@socket.io/redis-emitter@4`):

```js
const { Emitter } = require("@socket.io/redis-emitter");
const { createClient } = require("redis");

const redisClient = createClient();
const io = new Emitter(redisClient);
```
